src/drivers/gitlab.js

```diff
@@ -45,7 +72,7 @@ class Gitlab {
 
   async upload(opts = {}) {
     const { project_path, repo } = this;
-    const endpoint = `/projects/${project_path}/uploads`;
+    const endpoint = `${await this.detect_api_v4()}/projects/${project_path}/uploads`;
```

Comment:
Why not leave this and subsequent methods intact and recalculate the api_v4 within the request function ?

Reply:
Because we're also using the `request` function to detect the right address. Your suggested modification is rather elegant, but we would need to resolve this circular dependency somehow.

Reply:
You could pass endpoint as you where or url in opts. If endpoint is passed you call the detect_api... that is returning the proper one if its calculated if url is passed you just take the url

Reply:
@0x2b3bfa0

Reply:
Nice! 🙏